### PR TITLE
[MXAPPS-809] Configuration for Benchmark.AI Reporting.

### PR DESCRIPTION
**Pull request by vishaalkapoor**
[MXAPPS-809] Configuration for Benchmark.AI reporting.

The code will follow in a separate PR. This configuration will allow customers to onboard while the code is being PR'ed.

### CODE REVIEW

File: reporting/config/benchmarks.yaml [hunk 1]
```diff
@@ -0,0 +1,863 @@
...
+# TODO(vishaalk): Use default values to avoid redundancy.
+benchmarks:
+# ------------------ Tensorflow MKL --------------------------------------------
+  - Metric Prefix: mxnet.tensorflow_mkl_c5_resnet50_18xlg
+    Metric Suffix: nightly.c5_18x
+    Type: Training CV
+    Framework: Tensorflow
+    Model: ResNet-50
+    Benchmark: Batch size 1
+    Instance Type: C5.18xl
```
Comment (larroy) [inline reporting/config/benchmarks.yaml]: are these instance names custom abbreviations?

File: reporting/config/benchmarks.yaml [hunk 2]
```diff
@@ -0,0 +1,863 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#
+# Configuration for benchmarks to be included in the Benchmark.AI report. This file is of the following formmat:
```
Comment (larroy) [inline reporting/config/benchmarks.yaml]: formmat -> format

File: reporting/config/benchmarks.yaml [hunk 3]
```diff
@@ -0,0 +1,863 @@
...
+    Instance Type: C5.18xl
+    CPU Memory: cpu_memory_usage
+    Throughput: images/sec(training with batch size 1)
+
+  - Metric Prefix: mxnet.tensorflow_mkl_c5_resnet50_18xlg
+    Metric Suffix: nightly.c5_18x
+    Type: Training CV
+    Framework: Tensorflow
+    Model: ResNet-50
+    Benchmark: Batch size 32
```
Comment (larroy) [inline reporting/config/benchmarks.yaml]: What is this field?

File: reporting/config/benchmarks.yaml (hunk 1 above)
Comment (vishaalkapoor) [reply reporting/config/benchmarks.yaml]: The instance type will be pulled in verbatim into the report. These are abbreviations. I see your point, I can specify an enumeration in the configuration for the format to follow.

File: reporting/config/benchmarks.yaml (hunk 3 above)
Comment (vishaalkapoor) [reply reporting/config/benchmarks.yaml]: #      Benchmark: Optional Benchmark Description
I can name this more appropriately (Benchmark Desc:)

File: reporting/config/benchmarks.yaml (hunk 2 above)
Comment (vishaalkapoor) [reply reporting/config/benchmarks.yaml]: Done.

**Comment by vishaalkapoor** [comment]
Code following. 
The reason for splitting the config and code is to unblock those that want to add configuration and onboard to having reports. 

Pedro and I chatted offline about the code and config design, and will chat further to discuss how to best to handle the configuration. The issue is that the existing task configuration exists in many different personal GitHub repos, and so there isn't a central place that collects all official tests.

As an interim, we'll use this configuration file for the benchmarks that want to be added to a report. In the future we may move or merge this to a more appropriate place.

**Pull request merged**